src/bot.jl

```diff
@@ -1,55 +1,66 @@
+    BotConfig(package_name::AbstractString ; blacklist, os, else_os, version, else_version, package_path, precompiles_rootpath, subst, tmin)
 
-# Arguments:
-- `package_name::AbstractString`
+Construct a SnoopCompile bot configuration. `package_name` is the name of the package. This object is passed to [`snoop_bot`](@ref)
+and [`snoop_bench`](@ref).
 
 You may supply the following optional **keyword** arguments:
 
-- `blacklist` : A vector of of Strings (or RegExp) to remove some precompile sentences
+- `blacklist` : A vector of of Strings (or RegExp) to remove some precompile statements
 
-- `exhaustive`: passing `true` enables exhaustive testing of each precompilation signature, however, it slows down the processing stage. Defaults to `false`. Use this feature if SnoopCompile benchmark or your CI fails. This feature also prints the errors, which can help to find the `blacklist`s indivudally and disble `exhaustive` for the next runs.
+- `exhaustive`: passing `true` discards precompile statements that cannot be `eval`ed.
+  However, it slows down the processing stage.
+  Defaults to `false`. Use this feature if SnoopCompile benchmark or your CI fails.
+  This feature also prints the errors, which you can fix by `blacklist`ing the offending statements
+  and then disabling `exhaustive` for future runs.
 
-- `os`: A vector of of Strings (or RegExp) to give the list of os that you want to generate precompile signatures for. Each element will call a `Sys.is\$eachos()` function.
+- `os`: A vector of of Strings (or RegExp) to support with precompile statements.
+  Each element will call a `Sys.is\$eachos()` function.
 
 Example: `os = ["windows", "linux"]`
 
-- `else_os`: If you want to use a specific os for any other os, give `else_os` the name of that os. If you don't want to precompile for other os pass nothing.
+- `else_os`: If you want to use a specific operating system's precompile file as the default,
+  set `else_os` to the name of that os.
+  Setting this to `nothing` disables precompilation on any operating system other than those
+  explicitly listed in `os`.
 
 Example: `else_os = "linux"`
 
-- `version`: A vector of of versions to give the list of versions that you want to generate precompile signatures for.
+- `version`: A vector of of Julia versions used to generate precompile signatures.
 
 Example: `version = [v"1.1", v"1.4.1"]`
 
-It is assumed that the generated precompile signatures are valid for patch versions of Julia (e.g. giving v"1.4.2" assumses v"1.4.0" to v"1.4.9").
+It is assumed that the generated precompile signatures are valid for patch versions of Julia (e.g. giving v"1.4.2" supports v"1.4.0" and v"1.4.9").
 
-- `else_vresion`: If you want to use a specific version for any other version, give `else_version` the name of that version.
+- `else_version`: the Julia version used to generate the fallback (default) signatures.
```

Comment:
I am not sure what fallback means here
```suggestion
- `else_version`: the Julia version used to generate the default signatures.
Not passing this argument skips precompilation on any Julia version other than those explicitly listed in `version`.
```